pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -375,9 +375,10 @@ func (gc *GarbageCollector) attemptToDeleteItem(item *node) error {
 	default:
 		// item doesn't have any solid owner, so it needs to be garbage
 		// collected. Also, none of item's owners is waiting for the deletion of
-		// the dependents, so GC deletes item with Default.
-		glog.V(2).Infof("delete object %s with Default", item.identity)
-		return gc.deleteObject(item.identity, nil)
+		// the dependents, so GC deletes item in Background.
+		glog.V(2).Infof("delete object %s in Background", item.identity)
+		policy := metav1.DeletePropagationBackground
+		return gc.deleteObject(item.identity, &policy)
```

Comment:
Thanks, this is much better. I applied the change. I kept thinking about analyzing the finalizers of the cached object and worried about race but forgot the gc fetched `latest`. PTAL, @liggitt.